### PR TITLE
Implementing LoadingImageView

### PR DESCRIPTION
**Pull request by AnnaBanani**
Implementing LoadingImageView

### CODE REVIEW

**Review by Ssimboss — CHANGES_REQUESTED** [review]

File: TheMET/TheMET/Network/LoadingImageView.swift [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+//
+//  Created by Анна Ситникова on 28/04/2023.
+//
+
+import Foundation
+import UIKit
+
+class LoadingImageView: UIView {
+    
+    enum LoadingImageView {
```
[the remark this answers is not in the document]
Comment (AnnaBanani) [reply TheMET/TheMET/Network/LoadingImageView.swift:13]: fixed

File: TheMET/TheMET/Network/LoadingImageView.swift [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        case loading
+        case failed
+        case loaded(UIImage)
+    }
+    
+    private var state: LoadingImageView = .loaded(UIImage(named: "Not Loaded Image")!) {
```
[the remark this answers is not in the document]
Comment (AnnaBanani) [reply TheMET/TheMET/Network/LoadingImageView.swift:19]: fixed

File: TheMET/TheMET/Network/LoadingImageView.swift [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        self.failedImageView.contentMode = .center
+        NSLayoutConstraint.activate([
+            self.imageView.leadingAnchor.constraint(equalTo: self.leadingAnchor, constant: 0),
+            self.imageView.trailingAnchor.constraint(equalTo: self.trailingAnchor, constant: 0),
+            self.imageView.topAnchor.constraint(equalTo: self.topAnchor, constant: 0),
+            self.imageView.bottomAnchor.constraint(equalTo: self.bottomAnchor, constant: 0),
+            self.failedImageView.leadingAnchor.constraint(equalTo: self.leadingAnchor, constant: 0),
+            self.failedImageView.trailingAnchor.constraint(equalTo: self.trailingAnchor, constant: 0),
+            self.failedImageView.topAnchor.constraint(equalTo: self.topAnchor, constant: 0),
+            self.failedImageView.bottomAnchor.constraint(equalTo: self.bottomAnchor, constant: 0),
```
[the remark this answers is not in the document]
Comment (AnnaBanani) [reply TheMET/TheMET/Network/LoadingImageView.swift:58]: fixed

File: TheMET/TheMET/Network/LoadingImageView.swift [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    private func resetStateView(state: LoadingImageView) {
+        switch state {
+        case .loading:
+            self.imageView.image = nil
+            self.imageView.isHidden = true
+            self.failedImageView.isHidden = true
+            self.activityIndicator.isHidden = false
+            self.activityIndicator.startAnimating()
+        case .failed:
+            self.activityIndicator.stopAnimating()
```
[the remark this answers is not in the document]
Comment (AnnaBanani) [reply TheMET/TheMET/Network/LoadingImageView.swift:73]: fixed

**Review by Ssimboss — APPROVED** [review]

**Pull request merged**